File: src/main/resources/META-INF/resources/js/api/model/request/MessageRequest.js
/**
 * MessageRequest - matches com.mrngwozdz.api.model.request.MessageRequest
 */
class MessageRequest {
    constructor(text) {
        this.text = text;                    // String - The message text
    }

    toJSON() {
        return {
            text: this.text
        };
    }
}

// Export for use in other modules
if (typeof module !== 'undefined' && module.exports) {
    module.exports = MessageRequest;
} else if (typeof window !== 'undefined') {
    window.MessageRequest = MessageRequest;
}

File: src/main/java/com/mrngwozdz/api/model/request/MessageRequest.java
package com.mrngwozdz.api.model.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class MessageRequest {
    private String text;
    @JsonProperty("image_data")
    private byte[] imageData;
    @JsonProperty("image_filename")
    private String imageFilename;
}